src/components/providers.tsx

```diff
@@ -0,0 +1,45 @@
+/** @jsx jsx */
+import { jsx, ThemeProvider as ThemeUIProvider } from "theme-ui"
+import { ThemeProvider, getTheme } from "gatsby-interface"
+import { RunnerProvider } from "./site-runners"
+import { ReactNode } from "react"
+const baseTheme = getTheme()
+
+const theme = {
```

Comment:
Instead of the consistent spreads you can grab "merge" from theme-ui and use that. It'll deepmerge your baseTheme with the new stuff.